phoenix-core/src/main/java/org/apache/phoenix/schema/MetaDataClient.java

```diff
@@ -3300,6 +3281,134 @@ private void mutateStringProperty(String tenantId, String schemaName, String tab
         }
     }
 
+    public MutationState modifyColumn(ModifyColumnStatement statement) throws SQLException {
+        PTable table = FromCompiler.getResolver(statement, connection).getTables().get(0).getTable();
+        ColumnDef columnDef = statement.getColumnDef();
+        ColumnName columnName =columnDef.getColumnDefName();
+
+        // we can not modify index table/system table and project table.
+        if (table.getType() != PTableType.TABLE && table.getType() != PTableType.VIEW) {
```

Comment:
just check if the type is TABLE or VIEW

Reply:
what else type of tables should be modified? I think just TABLE and VIEW are be modified

Reply:
I misread this, ignore my previous comment.